hub/core/chunk_engine/write.py

```diff
@@ -149,13 +149,17 @@ def write_samples(
 
 
 def write_bytes(
```

Comment:
this PR is quite performant but the code is a bit hard to follow. Not sure how can that be improved.

Reply:
probably the function names

Reply:
No, there's definitely some higher-level restructuring that needs to happen here. It's extremely hard to follow without a mountain of context

Reply:
ok i will refactor

Reply:
i don't really understand what you guys don't understand. "needs higher-level restructuring" is really vague. what exactly isn't followable?